### PR TITLE
feat: error database if no rules (#2186)

### LINKED ISSUE #1450: Bootstrap stuck when rules.pb gets deleted
__Steps to reproduce:__

1. delete a rules.pb file in the object store for which the database directory has other data (e.g. parquet files)
2. restart iox

__Expected behaviour:__

IOx to boot successfully, knowing nothing about the existence of the deleted database.
Possibly warns that there is some data directory without a database config.

__Actual behaviour:__

```
'level=error msg="error getting database config AmazonS3(CloudPath { inner: Parsed(DirsAndFileName { directories: [PathPart(\"1\"), PathPart(\"844910ece80be8bc_infra\")], file_name: Some(PathPart(\"rules.pb\")) }) }) from object store: store error: AWS S3-based Object Store error: Unable to GET data. Bucket: toolsus1.a.influxcloud.net-iox, Location: 1/844910ece80be8bc_infra/rules.pb, Error: The specified key does not exist." target="server" location="server/src/lib.rs:476" time=1620334786114869837
```

repeats every `STORE_ERROR_PAUSE_SECONDS` (100) seconds.

### PR DESCRIPTION
**Pull request by tustvold**
#1456 added logic to not load databases with no rules.pb, from what I understand from the PR description this was a hack to fix #1450 where server startup would hang if it couldn't load a database.

I clarified with @pauldix that a "directory" under the server prefix that lacks a rules.pb file should be an error, whereas currently IOx just ignores the directory.

Now that database startup is separate from server startup it should be possible to remove this logic, without it preventing the server and unrelated databases from starting up correctly. 

Removing this logic feeds into the general server startup cleanup under #2186

### CODE REVIEW

File: server/src/lib.rs [hunk 1]
```diff
@@ -2097,6 +2102,11 @@ mod tests {
         let server_id = ServerId::try_from(1).unwrap();
         let db_name = "my_db";
 
+        // setup server
```
Comment (tustvold) [inline server/src/lib.rs:2105]: This has to be reordered as the existence of the catalog, creates the "directory" that the server uses to infer that a database exists at that location

**Review by crepererum — APPROVED** [review]
Looks good, however we must ensure that for FS-based stores we either don't list empty directories or that `store.delete(...)` wipes empty directories (this can be a separate PR). Otherwise deleting databases will be very hard to implement properly.

**Comment by tustvold** [comment]
>  Otherwise deleting databases will be very hard to implement properly.

Yes, I think deleting databases is something that will need a non-trivial amount of design work that doesn't currently exist. https://github.com/influxdata/influxdb_iox/issues/1881 looks to contain some initial discussion on this front

File: server/src/init.rs [hunk 2]
```diff
@@ -158,16 +158,11 @@ async fn initialize_database(
     )
     .await
     {
-        Ok(true) => {
+        Ok(_) => {
             // finished init and keep DB
             handle.commit();
             Ok(())
         }
-        Ok(false) => {
```
Comment (alamb) [inline server/src/init.rs:166]: This state also confused me in https://github.com/influxdata/influxdb_iox/pull/2180 - I am glad it is being removed 👍

**Review by alamb — APPROVED** [review]
I think this is a nice cleanup 👍

**Pull request merged**